src/metabase/api/common.clj

```diff
@@ -93,6 +95,21 @@
   []
   (check-403 *is-superuser?*))
 
+(defn check-has-general-permission
+  "If `advanced-permissions` is enabled, check `*current-user*` has general permission of type `perm-type`.
+  Set `require-superuser?` to `true` to perform a superuser check when `advanced-permissions` is disabled."
+  ([perm-type]
+   (check-has-general-permission perm-type true))
+
+  ([perm-type require-superuser?]
+   (classloader/require 'metabase.public-settings.premium-features
+                        'metabase.models.permissions)
+   (if ((resolve 'metabase.public-settings.premium-features/enable-advanced-permissions?))
+     (check-403 (or *is-superuser?*
+                    ((resolve 'metabase.models.permissions/set-has-general-permission-of-type)
```

Comment:
neither of these are EE namespaces, why do they need to be resolved?

Reply:
yes. it will fail due to cyclic dependencies...

Reply:
oh I see. maybe you can move this function to `metabase.api.common.validation`? that's a namespace I created recently when I ran into a similar cyclic dependency issue with `metabase.api.common`

Reply:
that's fanstatic :)